server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -143,6 +147,95 @@ function getTokenFromRequest(request: Request): string {
 	return tokenMatch[1];
 }
 
+export async function verifyToken(
+	tenantId: string,
+	documentId: string,
+	token: string,
+	tenantManager: ITenantManager,
+	options: IVerifyTokenOptions,
+): Promise<void> {
+	if (options.requireDocumentId && !documentId) {
+		throw new NetworkError(403, "Missing documentId.");
+	}
+
+	let tokenLifetimeMs: number | undefined;
+	try {
+		const claims = validateTokenClaims(token, documentId, tenantId, options.requireDocumentId);
+		if (options.requireTokenExpiryCheck) {
+			if (!options.maxTokenLifetimeSec) {
```

Comment:
I'm wondering if we should just log this error but not throw on it. This is a misconfiguration error, so then all Auth would receive 500 errors which would essentially mean no one could access the service because we messed up token lifetime config.

A different option would be to move this check to where these options are read from the config, which while we're at it could be moved out of the per-request handler. That would cause the deployment to fail and be rolled back due to service boot error, which I think is preferable for this case.

Reply:
Like so
```ts
export function verifyStorageToken(
	tenantManager: ITenantManager,
	config: Provider,
	tokenManager: ITokenRevocationManager | undefined,
	options: IVerifyTokenOptions = {
		requireDocumentId: true,
		ensureSingleUseToken: false,
		singleUseTokenCache: undefined,
		enableTokenCache: false,
		tokenCache: undefined,
	},
): RequestHandler {
	const maxTokenLifetimeSec = config.get("auth:maxTokenLifetimeSec") as number;
	const isTokenExpiryEnabled = config.get("auth:enableTokenExpiration") as boolean;
	if (isTokenExpiryEnabled && !maxTokenLifeTimeSec) {
		// Prevent service from starting with invalid configs
		throw new Error("Invalid configuration: no maxTokenLifetimeSec when token expiry is enabled");
	}
	return async (request, res, next) => {
		const tenantId = getParam(request.params, "tenantId");
		// ...
```

Reply:
Yet another option would be to add a default maxTokenLifetimeSec of 1hr

Reply:
Not sure the differences of placing this check in verifyToken vs verifyStorageToken since they will be only triggered when a request come. I think it won't be caught by service boot stage. But I agree that we don't want this misconfig to block the service. I will use the default value approach and log an error instead.

Reply:
> Not sure the differences of placing this check in verifyToken vs verifyStorageToken since they will be only triggered when a request comes

This is incorrect. If you look at my example code, you can see that it's checked _outside_ of the individual request wrapper. This will throw an error on service instantiation. The way you are doing it right now won't. DM me on Teams if you need me to explain more about middlewares and request handlers

Reply:
I think I get your point. verifyStorageToken is executed in function router.get (as an argument) during service startup and by moving code out of the returned middleware, the check code is executed during the start up stage. I added the check there as well as within the verifyToken function since the maxTokenLife could be undefined inside the function. From the function's perspective, it should check the input first. Thanks for the suggestions